projects/element-ng/loading-spinner/si-loading-spinner.directive.ts

```diff
@@ -86,12 +86,19 @@ export class SiLoadingSpinnerDirective implements OnInit, OnChanges, OnDestroy {
         useValue: true
       }
     ];
-    const outlet = new DomPortalOutlet(
-      this.el.nativeElement,
-      undefined,
-      this.appRef,
-      Injector.create({ providers })
-    );
+
+    // Ensure compatibility between Angular CDK 19 and 20 since constructor arguments have changed
+    const outlet =
+      DomPortalOutlet.length > 3
+        ? new DomPortalOutlet(
+            this.el.nativeElement,
+            undefined,
+            this.appRef,
+            Injector.create({ providers })
+          )
+        : // @ts-expect-error: DomPortalOutlet constructor signature with CDK 20
+          new DomPortalOutlet(this.el.nativeElement, this.appRef, Injector.create({ providers }));
```

Comment:
instead of doing this I would do like following:

```
private viewRef = inject(ViewContainerRef);
private compPortal = new ComponentPortal(SiLoadingSpinnerComponent, this.viewRef, Injector.create({
    providers: [
      { provide: LOADING_SPINNER_BLOCKING, useValue: this.blocking() },
      {
        provide: LOADING_SPINNER_OVERLAY,
        useValue: true
      }
    ]
}));

new DomPortalOutlet(this.el.nativeElement);
```